src/imap/mod.rs

```diff
@@ -1085,6 +1085,24 @@ impl Imap {
                         return ImapActionResult::AlreadyDone;
                     };
 
+                    if let Some(fetched_uid) = fetch.uid {
+                        if fetched_uid != uid {
+                            warn!(
+                                context,
+                                "IMAP bug: fetched UID {} is not equal to requested UID {}",
+                                fetched_uid,
+                                uid
+                            );
+                            return ImapActionResult::RetryLater;
+                        }
+                    } else {
+                        warn!(
+                            context,
+                            "IMAP bug: fetch result doesn't contain requested UID {}", uid
+                        );
+                        return ImapActionResult::RetryLater;
+                    }
+
```

Comment:
if this is becaue of a bug in async-imap there needs to be a comment to that effect.

Reply:
If this is a persistent error, it will lead to retrying each message deletion quite a lot. 
Do you have indication the issue is temporary?  Could the UID validity have changed?

Reply:
It doesn't matter if it is a bug in async-imap or in the server. The condition checks that in response to UID FETCH 100 (UID ...) we get UID 100, not something else.

I think it is async-imap that gets confused because it did not read a response once, so it reads old response after requesting another one, but it is yet to be determined.

Reply:
I added `trigger_reconnect` to ensure IMAP recovers from this condition.